### PR TITLE
util/chunk: migrate test-infra to testify for all remaining tests

### LINKED ISSUE #26170: migrate test-infra to testify for `util/chunk` pkg

### PR DESCRIPTION
**Pull request by xxchan**
### What problem does this PR solve?

Issue Number: part of https://github.com/pingcap/tidb/issues/26170, 
includes codec_test, row_container_test, pool_test, list_test, mutrow_test

What's Changed:
- **Only trivial regexp change**
- [x] make sure "github.com/pingcap/check"  is removed, thus closes #26170

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by tisonkun** [comment]
@xxchan it looks the previous PRs merged, may we revive this one.

File: util/chunk/chunk_test.go [hunk 1]
```diff
@@ -1021,7 +1012,7 @@ func benchmarkChunkGrow(t benchChunkGrowCase) func(b *testing.B) {
 	}
 }
 
-func (s *testChunkSuite) TestAppendRows(c *check.C) {
+func TestAppendRows(t *testing.T) {
```
Comment (xhebox) [inline util/chunk/chunk_test.go:1015]: Parallel all the tests maybe?

File: util/chunk/row_container_test.go [hunk 2]
```diff
@@ -102,54 +102,56 @@ func (r *rowContainerTestSuite) TestSpillAction(c *check.C) {
...
 	}
 
 	err = rc.Reset()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 }
 
-func (r *rowContainerTestSerialSuite) TestSpillActionDeadLock(c *check.C) {
+func TestSpillActionDeadLock(t *testing.T) {
+	t.Parallel()
```
Comment (xhebox) [inline util/chunk/row_container_test.go:144]: Serial test, plz don't parallel it.

File: util/chunk/row_container_test.go [hunk 3]
```diff
@@ -200,35 +206,35 @@ func (r *rowContainerTestSuite) TestSortedRowContainerSortSpillAction(c *check.C
...
-	c.Assert(errors.Is(err, ErrCannotAddBecauseSorted), check.IsTrue)
+	require.Error(t, err)
+	require.ErrorIs(t, err, ErrCannotAddBecauseSorted)
 	err = rc.Reset()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 }
 
-func (r *rowContainerTestSerialSuite) TestActionBlocked(c *check.C) {
+func TestActionBlocked(t *testing.T) {
```
Comment (xhebox) [inline util/chunk/row_container_test.go:235]: ditto

File: util/chunk/row_container_test.go (hunk 2 above)
Comment (xxchan) [reply util/chunk/row_container_test.go:144]: What makes it serial?

Comment (xhebox) [reply util/chunk/row_container_test.go:144]: rowContainerTestSerialSuite, in this test, it is failpoint.

Comment (xxchan) [reply util/chunk/row_container_test.go]: Thank you. Resolved.

File: util/chunk/row_container_test.go [hunk 4]
```diff
@@ -102,54 +102,54 @@ func (r *rowContainerTestSuite) TestSpillAction(c *check.C) {
...
+		require.Equal(t, chk.GetRow(rowIdx).GetDatumRow(fields), resChk.GetRow(rowIdx).GetDatumRow(fields))
 	}
 
 	err = rc.Reset()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 }
 
-func (r *rowContainerTestSerialSuite) TestSpillActionDeadLock(c *check.C) {
+func TestSpillActionDeadLock(t *testing.T) {
```
Comment (tisonkun) [inline util/chunk/row_container_test.go:143]: such as this test

File: util/chunk/row_container_test.go [hunk 5]
```diff
@@ -200,35 +204,33 @@ func (r *rowContainerTestSuite) TestSortedRowContainerSortSpillAction(c *check.C
...
-	c.Assert(errors.Is(err, ErrCannotAddBecauseSorted), check.IsTrue)
+	require.Error(t, err)
+	require.ErrorIs(t, err, ErrCannotAddBecauseSorted)
 	err = rc.Reset()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 }
 
-func (r *rowContainerTestSerialSuite) TestActionBlocked(c *check.C) {
+func TestActionBlocked(t *testing.T) {
```
Comment (tisonkun) [inline util/chunk/row_container_test.go:233]: and this one.

**Review by tisonkun — COMMENTED** [review]
@xhebox I'd prefer move serial tests into same file(s), such as `xxx_serial_test.go`, otherwise it is brittle.

File: util/chunk/row_container_test.go (hunk 4 above)
Comment (xxchan) [reply util/chunk/row_container_test.go]: done

File: util/chunk/row_container_serial_test.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 PingCAP, Inc.
```
[the remark this answers is not in the document]
Comment (xxchan) [reply util/chunk/row_container_serial_test.go:1]:
```suggestion
// Copyright 2021 PingCAP, Inc.
```

File: util/chunk/row_container_test.go [hunk 7]
```diff
@@ -200,84 +166,35 @@ func (r *rowContainerTestSuite) TestSortedRowContainerSortSpillAction(c *check.C
...
-		c.Assert(row.GetInt64(0), check.Equals, int64(i/2))
+		require.NoError(t, err)
+		require.Equal(t, int64(i/2), row.GetInt64(0))
 	}
 	// Can't insert records again.
 	err = rc.Add(chk)
-	c.Assert(err, check.NotNil)
-	c.Assert(errors.Is(err, ErrCannotAddBecauseSorted), check.IsTrue)
+	require.Error(t, err)
+	require.ErrorIs(t, err, ErrCannotAddBecauseSorted)
```
[the remark this answers is not in the document]
Comment (xxchan) [reply util/chunk/row_container_test.go:190]: I think it is not equivalent: e.g., if `error` interface is implemented using a pointer receiver, `errors.Is((*MyErr)(nil), target)` will return true.

**Review by mmyj — APPROVED** [review]
lgtm

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
@mmyj could you help on merging this PR?

**Pull request merged**